scripts/pi-hole/lua/footer.lp

```diff
@@ -6,6 +6,16 @@
 *  This file is copyright under the latest version of the EUPL.
 *  Please see LICENSE file for your rights under this license.
 * ]]--
+
+-- Iterate over the array mg.request_info.http_headers and check if there is an 'X-Forwarded-For' header
+x_forwarded_for = ""
+local headers = mg.request_info.http_headers
+for k,v in pairs(headers) do
```

Comment:
I know this is still a draft, but I have a few questions:

Why are you creating a variable here if this variable is never reused? Why not use `mg.request_info.http_headers` directly?
Why use `local`? Is there a reason? 

```suggestion
for k,v in pairs(mg.request_info.http_headers) do
```

Reply:
> Why are you creating a variable here if this variable is never reused? Why not use `mg.request_info.http_headers` directly?

This is how one typically does it in LUA to keep code more readable, it will not create a copy but instead work with the pointer to the same LUA object.

> Why use local? Is there a reason?

`local` defines the variable locally to the "chunk". In LUA server pages, a chunk starts at `<?` and ends at `?>`, i.e., it isn't defined everywhere in the script but really only `local`ly.

Note that `local` can be a bit counter-intuitive as "chunks" aren't always clear, e.g. if you run LUA in interactive mode, each statement will be it's own chunk and `i` will already be out-of-scope once you enter the `print(i)` in this example:
``` lua
> local i=1
> print(i)
nil
```
Nonetheless, when everything is in one "chunk", it will work:
``` lua
> local i=1; print(i)
1
```

Reply:
I removed the intermediate variable now

Reply:
I think if you enclose these lines in a `do ... end` block it will be considered a separate "chunk", but I don't think this is needed here.